wcfsetup/install/files/lib/data/user/UserProfile.class.php

```diff
@@ -103,6 +103,12 @@ class UserProfile extends DatabaseObjectDecorator implements ITitledLinkObject {
 	 */
 	protected $coverPhoto;
 	
+	/**
+	 * @var Trophy[]
+	 * @since 5.3
+	 */
+	protected $specialTrophies;
```

Comment:
Saving this in a class variable causes problems when automatically marking special trophies. Currently, this is read correctly each time and if the value changes, the current value in each object is updated. Not anymore with this update. So here you should either a) continue to use the object from the UserStorageHandler (does deserialization really generate such a large overhead?) or b) add a parameter $skipCache which skips the cache (and use it in the UserTrophyAction::create).

Reply:
@BurntimeX